samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -219,6 +220,11 @@ public void run() {
       //initialize JobCoordinator state
       LOG.info("Starting cluster based job coordinator");
 
+      // write the diagnostics metadata file
+      Optional<String> execEnvContainerId = Optional.ofNullable(System.getenv("CONTAINER_ID"));
```

Comment:
Is there a better way of obtaining this?
In AM the yarn-containerID is in env variable CONTAINER_ID whereas on containers it is under 
EXEC_ENV_CONTAINER_ID.